arroyo-api/src/pipelines.rs

```diff
@@ -308,31 +311,34 @@ pub(crate) async fn create_pipeline<'a>(
         }
     };
 
-    optimizations::optimize(&mut compiled.program.graph);
+    // TODO: graph optimizations?
```

Comment:
I think we should be able to move these all into the various datafusion optimization passes.